genesis/public/core/reference.go

```diff
@@ -33,13 +33,16 @@ type ReferenceDomain interface {
 	// Base domain implementation.
 	domain.Domain
 	// RegisterReference is used to publish new global references
-	RegisterReference(*object.Reference) (string, error)
+	RegisterReference(*object.Reference, string) (string, error)
 	// ResolveReference provides reference instance from record
 	ResolveReference(string) (*object.Reference, error)
+	// SetMap set globalResolverMap for references register/resolving
+	SetMap(globalInstanceMap *map[string]object.Proxy) error
```

Comment:
`SetMap` is not good context name here. Maybe `InitGlobalResolver` is better?

Reply:
Its not about init global resolver, its about set inner reference domain map at the same memory point as global resolver map. InitGlobalMap mb?